modules/relation/src/main/Env.scala

```diff
@@ -40,12 +41,17 @@ final class Env(
     maxBlock = MaxBlock)
 
   val onlinePlayings = new lila.memo.ExpireSetMemo(4 hour)
+  val onlineStudying = new OnlineStudyingMemo(4 hour) // people with write access in public studies
+  val onlineStudyingAll = new OnlineStudyingMemo(4 hour) // people with write or read access in public and private studies
```

Comment:
`4 hours` was for 45+45 games, since playing moves doesn't refresh the cache. I think `20 minutes` of inactivity should be more than enough for studies.